zcash_client_sqlite/src/wallet/init.rs

```diff
@@ -558,6 +559,33 @@ mod tests {
         }
 
         let expected_views = vec![
+            // v_sapling_shard_unscanned_ranges
+            format!(
+                "CREATE VIEW v_sapling_shard_unscanned_ranges AS
+                SELECT
+                    shard.shard_index,
+                    shard.shard_index << 16 AS start_position,
+                    (shard.shard_index + 1) << 16 AS end_position_exclusive,
+                    IFNULL(prev_shard.subtree_end_height, {}) AS subtree_start_height,
+                    shard.subtree_end_height AS subtree_end_height,
+                    shard.contains_marked,
+                    scan_queue.block_range_start,
+                    scan_queue.block_range_end,
+                    scan_queue.priority
+                FROM sapling_tree_shards shard
+                LEFT OUTER JOIN sapling_tree_shards prev_shard
+                    ON shard.shard_index = prev_shard.shard_index + 1
+                INNER JOIN scan_queue ON 
+                    (scan_queue.block_range_start BETWEEN subtree_start_height AND shard.subtree_end_height) OR
+                    (scan_queue.block_range_end BETWEEN subtree_start_height AND shard.subtree_end_height) OR
+                    (
+                        scan_queue.block_range_start <= prev_shard.subtree_end_height
+                        AND scan_queue.block_range_end >= shard.subtree_end_height
```

Comment:
This could be `>` because the `==` case is caught by the previous `BETWEEN` clause. It's correct as-is (assuming inclusive ranges) though.

Reply:
There are three cases that we're concerned with: the block range overlaps the start of the subtree range, the block range overlaps the end of the subtree range, or the block range fully contains the subtree range. I thought expressing it this way was clearest, because you can verify that each case is independently correct.